Adds release notes for image loading iteration I merged changes

JorgeMucientes opened the pull request:
### Description
Just updates release notes including details for all of the merged changes related to Iteration I

JorgeMucientes committed:
Adds release notes for image loading iteration I merged changes

dangermattic commented:
<!--
  0 Errors
  1 Warning: This PR is assigned to the mil...
  0 Messages
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/woocommerce/woocommerce-android/milestone/249">21.6</a>. The due date for this milestone has already passed.<br />
Please assign it to a milestone with a later deadline or check whether the release for this milestone has already been finished.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

hafizrahman committed:
Remove redundant release note (already added in 21.5)

Additionally, the item "Optimizes product image fetching for faster loading" was already mentioned in 21.5 but as an internal note. I think it's OK to add it again and make it public.

hafizrahman commented:
Added a commit to update the notes.

> Remove redundant release note (already added in 21.5)
> Additionally, the item "Optimizes product image fetching for faster loading" was already mentioned in 21.5 but as an internal note. I think it's OK to add it again and make it public.

hafizrahman reviewed (approved):
LGTM
